CS4125Project/src/MenuUI/CheckoutUI.java

```diff
@@ -192,8 +193,12 @@ private void jButton5ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRS
             dispatcher.dispatch(new PurchaseRequest(p));
         });        
        
-        ShoppingBasket.GetInstance().ClearBasket();
         dispose();
+        if(Shop.getInstance().getAccount().getPaymentType() != null){
```

Comment:
Is there any need for the 'this.setVisible(false)'?
The window is already being terminated via the 'dispose()' call.